src/lib/ifaces/mac_vlan.rs

```diff
@@ -0,0 +1,164 @@
+use crate::mac::parse_as_mac;
+use crate::netlink::parse_as_u16;
+use crate::netlink::parse_as_u32;
+use crate::Iface;
+use crate::IfaceType;
+use netlink_packet_route::rtnl::link::nlas;
+use netlink_packet_route::rtnl::nlas::NlasIterator;
+use serde_derive::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+const ETH_ALEN: usize = 6;
+
+const MACVLAN_MODE_PRIVATE: u32 = 1;
+const MACVLAN_MODE_VEPA: u32 = 2;
+const MACVLAN_MODE_BRIDGE: u32 = 4;
+const MACVLAN_MODE_PASSTHRU: u32 = 8;
+const MACVLAN_MODE_SOURCE: u32 = 16;
+
+const IFLA_MACVLAN_MODE: u16 = 1;
+const IFLA_MACVLAN_FLAGS: u16 = 2;
+// const IFLA_MACVLAN_MACADDR_MODE: u16 = 3;    // not used in query
+const IFLA_MACVLAN_MACADDR: u16 = 4;
+const IFLA_MACVLAN_MACADDR_DATA: u16 = 5;
+const IFLA_MACVLAN_MACADDR_COUNT: u16 = 6;
+
+#[serde(rename_all = "lowercase")]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum MacVlanMode {
+    /* don't talk to other macvlans */
+    Private,
+    /* talk to other ports through ext bridge */
+    Vepa,
+    /* talk to bridge ports directly */
+    Bridge,
+    /* take over the underlying device */
+    #[serde(rename = "passthru")]
+    PassThrough,
+    /* use source MAC address list to assign */
+    Source,
+    Other(u32),
+    Unknown,
+}
```

Comment:
Should we specify what is doing each mode in comments? I mean, if we do this, we should do it with other interfaces as well to be consistent. Thanks!

Reply:
Just copy a kernel comment. And someday we can convert them into better documentation using `//`